modules/postgres/src/main/scala/doobie/postgres/syntax/FragmentSyntax.scala

```diff
@@ -23,14 +28,39 @@ class FragmentOps(f: Fragment) {
     // rows is an error so we shortcut on empty input.
     // TODO: stream this rather than constructing the string in memory.
     if (fa.isEmpty) 0L.pure[ConnectionIO] else {
-      val data = fa.foldLeft(new StringBuilder) { (b, a) =>
-        ev.unsafeEncode(a, b)
-        b.append("\n")
-      } .toString
+      val data = foldToString(fa)
       PHC.pgGetCopyAPI(PFCM.copyIn(f.query.sql, new StringReader(data)))
     }
   }
 
+  /**
+   * Given a fragment of the form `COPY table (col, ...) FROM STDIN` construct a
+   * `ConnectionIO` that inserts the values provided by `stream`, returning the number of affected
+   * rows. Chunks input `stream` for more efficient sending to `STDIN` with `minChunkSize`.
+   */
+  def copyIn[F[_]: ConcurrentEffect, A: Text](
+    stream: Stream[F, A],
+    minChunkSizeBytes: Int
```

Comment:
@tpolecat I'm not quite sure about `Bytes` part in the name, since it can be misleading. We chunk `Stream` of `A` before converting to string and then to bytes.